tools/provenance-generator/kn-attestation.json

```diff
@@ -0,0 +1,56 @@
+{
+    "_type": "https://in-toto.io/Statement/v0.1",
+    "predicateType": "https://slsa.dev/provenance/v0.2",
+    "subject": [
+      {
+        "name": "gcr.io/knative-releases/knative.dev/serving/cmd/controller",
+        "digest": {
+          "sha256": "bac158dfb0c73d13ed42266ba287f1a86192c0ba581e23fbe012d30a1c34837c"
+        }
+      },
+      {
+        "name": "gcr.io/knative-releases/knative.dev/serving/cmd/queue",
+        "digest": {
+          "sha256": "83f6888ea9561495f67334d044ffa8ad067d251ad953358dda7ea5183390cc69"
+        }
+      }
+    ],
+    "predicate": {
+      "builder": {
+        "id": "https://prow.knative.dev"
+      },
+      "buildType": "https://cloudbuild.googleapis.com/CloudBuildYaml@v1",
+      "invocation": {
+        "configSource": {
+          "entryPoint": "https://github.com/knative/serving/blob/main/hack/release.sh"
+        },
+        "parameters": [
+          "runner.sh",
+          "./hack/release.sh",
+          "--auto-release",
+          "--release-gcs=knative-releases/async-component",
+          "--release-gcr=gcr.io/knative-releases",
+          "--github-token=/etc/hub-token/token"
+        ]
```

Comment:
How is the build triggered? This section should capture the details of whatever triggered the build.

I think the parameters you need to create the resulting `buildConfig` are captured in the [container `ENTRYPOINT_OPTIONS`](https://storage.googleapis.com/knative-prow/logs/nightly_net-istio_main_periodic/1548602272080990208/podinfo.json#:~:text=%22name%22%3A%20%22ENTRYPOINT_OPTIONS%22%2C), which is great because you don't have to figure out how to capture them at the start of the job:

```json
{
  "name": "ENTRYPOINT_OPTIONS",
  "value": "{\"timeout\":7200000000000,\"grace_period\":15000000000,\"artifact_dir\":\"/logs/artifacts\",\"args\":[\"runner.sh\",\"./hack/release.sh\",\"--publish\",\"--tag-release\"],\"container_name\":\"test\",\"process_log\":\"/logs/process-log.txt\",\"marker_file\":\"/logs/marker-file.txt\",\"metadata_file\":\"/logs/artifacts/metadata.json\"}"
}
```

The goal of these sections of the provenance attestation is to express something like:

>  "prow (builder id) did (invocation) this (buildConfig) to the source (materials) and as a result we now have these artifacts (the subjects)"